test/gtest/uct/test_peer_failure.cc

```diff
@@ -56,10 +56,25 @@ class test_uct_peer_failure : public uct_test {
         return UCS_OK;
     }
 
+    typedef struct {
+        uct_pending_req_t    uct;
+        uct_ep_h             ep;
+    } pending_send_request_t;
+
     static ucs_status_t pending_cb(uct_pending_req_t *self)
     {
-        m_req_count++;
-        return UCS_OK;
+        const uint64_t send_data    = 0;
+        pending_send_request_t *req = ucs_container_of(self,
+                                                       pending_send_request_t,
+                                                       uct);
+
+        ucs_status_t status;
+        do {
+            status = uct_ep_am_short(req->ep, 0, 0, &send_data,
```

Comment:
isn't one try enough? then you can just return status

Reply:
no, we need to make sure that all pending requests won't be completed from the progress (e.g. 
 they are completed in `flush()` for RC_MLX5 transport). we want to purge them from the pending queue

Reply:
you just want to have pending reqs when calling purge, right?
so you execute communication operation to make sure that not all pending ops will be dispatched. 
Why do you need to wait for UCS_OK?

Reply:
> Why do you need to wait for UCS_OK?

this is waiting for the error, not for the `UCS_OK`

Reply:
why?

Reply:
> why?

to not remove the UCT requests from the pending queue and will be able to purge them from at the end of the testing (i.e. test `uct_ep_pending_purge()`).
In any words, if we do AM Short and it is completed with `UCS_OK`, then it will be deleted from the UCT pending queue.

Reply:
aa ok, so you will block in one send rather than issuing many

Reply:
> aa ok, so you will block in one send rather than issuing many

yes, this explains better what I wanted

Reply:
done